st-embroider/2023/2003-01-23.md

```diff
@@ -0,0 +1,130 @@
+  - the define is being driven by data that is coming out of the splitter
+  - you will likely need to add the correct implementation to the `webpack.ts` entryTemplate
+  - in the issue there is a missing define, the define example in the issue is showing it as a consumer so something is going to need to define what 
+  - (discussion about other odd things in the build - a bit of investigation)
+  - slightly trolly thing: you need to add webpack config to autoImport **and** the embroider config in your ember-auto-import
+  - if we want to allow people to import things from their own app in ember-auto-import then we will need to implement the first feature we discussed today
+  - re: trolly webpack config there are a few options that we can do. open to suggestions to make it better
+<5m> should ember-auto-import be under the embroider org? (related to release conversation and bus number) <@mansona>
+  - maybe
+  - CM: this is about allowing more peopel to release the important things
+  - EF: sure sounds good
+  - EF: we could move it to the embroider monorepo
+  - CM: I wasn't suggesting that 🙈
+<2m> ember-cli-babel v8 https://github.com/embroider-build/embroider/pull/1334 <@nullvoxpopuli>
+  - (discussion about the new semantics of ember-cli-babel)
+  - consensus: let's just merge this
+  - (more discussion)
+  - CM: let's comment the changes and we can get this in the next release. Releasing is more urgent
+  - EF: sure
+
+Do a release
+  - CM: can we record it
+  - EF: probably written is better
+  - CM: I will do that with the notes
+
+
+  process: 
+    - I look at the changelog and find the release name
+    - find the corresponding tag
+    - I look at the git log and search for "Merge pull" and make sure that all pull requests are covered
+    - (trying to remember changesets)
+   
+    - asside there is a lot of manual work
+    - NVP: that's because we forgot to add changeset to PRS before merging
+    - CM: can't make that the PR authors job, and can't put it on the maintainer
+    - EF: I need a list of all changes per package so we know what packages need to be released
+    - EF: with a monorepo that we have we don't want per-package changelogs
+    - the big thing is that it can't generate the changelog for me
+    - reference: there is a quest issue about this
+    
+    - create the new release in the markdown
+
+    ```
+    # Release 2022-11-30.0
+    ```
+
+    - multi pass process
+
+    - get PR 
+    - find what files/packages changed - > ad a header for the package and if it is a patch/minor/major
+    - add pr under the header
+
+    ```
+    ## `@embroider/compat`, `@embroider/core`, `@embroider/test-setup`, `@embroider/webpack` patch
+
+    - BUGFIX: detect ember-template-compliation plugin correctly when other plugins are preventing parallelization [1299](https://github.com/embroider-build/embroider/pull/1299)
+    ```
+
+    - I'm going to keep these alphabetical by package
```

Comment:
- run `npx changeset` to generate one file

	- manually edit ./changesets folder and duplicate files as needed to match the packages previously identified in updating the changelog manually

	-run `npx changeset version` which deletes the files in .changesets and updates versions

	- check what changesets did to version numbers

	- add version numbers `from -> to` eg `2.0.2 => 2.1.0` into CHANGELOG.md based on changes from previous changesets version command

	- commit everything and tag eg `release-2023-01-24.0`

	- make sure things are really clean: fresh install of node modules
	which does a fresh compile of all ts files (to avoid weird things from debugging)

	- release the packages one by one by going into the respective folders and doing npm publish
		- consider if you need any arguments like doing a beta release
		- add `npm publish --access=public` (it's safe so you don't have to think about it)
		- continue with all other packages mentioned in the CHANGELOG.md

	- now wait for bug reports to roll in